homeassistant/components/fully_kiosk/services.py

```diff
@@ -55,6 +58,17 @@ async def collect_coordinators(
             coordinators.append(hass.data[DOMAIN][config_entry.entry_id])
         return coordinators
 
+    async def async_change_location(call: ServiceCall) -> None:
+        """Open relative URL on the Fully Kiosk Browser without reloading page."""
+
+        # Encode string to prevent accidental JavaScript execution
+        relative_url = json.dumps(call.data[ATTR_URL])
+        javascript_url = f"""
+        javascript:history.pushState(null,"",{relative_url});window.dispatchEvent(new CustomEvent("location-changed"));
```

Comment:
This is too much protocol details. Those should be kept in the 3re party library.

Reply:
I'm not sure that the 3rd party library is the right place for this. There's nothing about this that is specific to FKB.

Most (if not all) of the FKB integration users would use this to be able to change HA dashboards on a kiosk without reloading the HA UI. So this isn't a FKB issue per se.

Do you have an opinion on this @cgarwood ?

Reply:
I agree with Martin on this. This is not something we should add / should accept in Home Assistant itself.

../Frenck